src/v/storage/segment_set.cc

```diff
@@ -280,6 +278,20 @@ unsafe_do_recover(segment_set&& segments, ss::abort_source& as) {
             if (unlikely(as.abort_requested())) {
                 return segment_set(std::move(good));
             }
+
+            if (
+              last_clean_segment
+              && std::filesystem::path(s->filename()).filename().string()
```

Comment:
do you think there is any reason to either use some std::filesystem::path normalization? for example I don't know where all the paths come from. could I use a `root` dir that contains `../../` and mess up this logic? alternatively, I could see using the metadata from the file name instead, for example (e.g. `struct { offset, term };`).

Reply:
Originally I used full paths, but that was wasteful + subject to the kinds of path/symlink aliasing that you're alluding do: that's why the `std::filesystem::path(...).filename()` is used to take just the basename.

It would be neat to just use the offset+term, although kind of awkward from the call-site where we use it (I think segment currently doesn't get knowledge of its term outside of the filename, and it feels a bit gross to be parsing out the filename during close())

Reply:
I missed the `.filename()` call. Since this the key is ntp, this seems like a robust normalization.